File: packages/backend/app/Models/FormField.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class FormField extends Model
{
    use HasFactory, HasUuids;

    /**
     * The primary key type.
     */
    protected $keyType = 'string';

    /**
     * Indicates if the IDs are auto-incrementing.
     */
    public $incrementing = false;

    /**
     * The attributes that are mass assignable.
     */
    protected $fillable = [
        'form_definition_id',
        'field_name',
        'field_type',
        'field_selector',
        'field_purpose',
        'preset_value',
        'is_sensitive',
        'is_file_upload',
        'is_required',
        'options',
        'sort_order',
    ];

    /**
     * Get the attributes that should be cast.
     */
    protected function casts(): array
    {
        return [
            'is_sensitive' => 'boolean',
            'is_file_upload' => 'boolean',
            'is_required' => 'boolean',
            'options' => 'array',
        ];
    }

    /**
     * Get the form definition that owns the form field.
     */
    public function formDefinition(): BelongsTo
    {
        return $this->belongsTo(FormDefinition::class);
    }
}

File: packages/scraper/app/models/form_field.py
from sqlalchemy import Column, String, Text, Boolean, Integer, DateTime, ForeignKey, JSON
from sqlalchemy.dialects.postgresql import UUID
from app.database import Base
import uuid


class FormField(Base):
    __tablename__ = 'form_fields'

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    form_definition_id = Column(UUID(as_uuid=True), ForeignKey('form_definitions.id'), nullable=False)
    field_name = Column(String(255), nullable=False)
    field_type = Column(String(50), nullable=False)
    field_selector = Column(Text, nullable=False)
    field_purpose = Column(String(100), nullable=True)
    preset_value = Column(Text, nullable=True)
    is_sensitive = Column(Boolean, default=False)
    is_file_upload = Column(Boolean, default=False)
    is_required = Column(Boolean, default=False)
    options = Column(JSON, nullable=True)
    sort_order = Column(Integer, default=0)
    created_at = Column(DateTime)
    updated_at = Column(DateTime)
